test/Cids.t.sol

```diff
@@ -43,4 +43,96 @@ contract CidsTest is Test {
         vm.expectRevert("Cid data is too short");
         Cids.digestFromCid(c);
     }
+
+    function testUvarintLength() public pure {
+        assertEq(Cids._uvarintLength(0), 1);
+        assertEq(Cids._uvarintLength(1), 1);
+        assertEq(Cids._uvarintLength(127), 1);
+        assertEq(Cids._uvarintLength(128), 2);
+        assertEq(Cids._uvarintLength(16383), 2);
+        assertEq(Cids._uvarintLength(16384), 3);
+        assertEq(Cids._uvarintLength(2097151), 3);
+        assertEq(Cids._uvarintLength(2097152), 4);
+        assertEq(Cids._uvarintLength(type(uint256).max), 37);
+    }
+
+    function testUvarintRoundTrip() public pure {
+        uint256[] memory values = new uint256[](7);
+        values[0] = 0;
+        values[1] = 1;
+        values[2] = 127;
+        values[3] = 128;
+        values[4] = 16384;
+        values[5] = 2097152;
+        values[6] = type(uint256).max;
+
+        uint256 totalLength = 0;
+        for (uint256 i = 0; i < values.length; i++) {
+            totalLength += Cids._uvarintLength(values[i]);
+        }
+        bytes memory buffer = new bytes(totalLength);
+        uint256 offset = 0;
+
+        // Write all values
+        for (uint256 i = 0; i < values.length; i++) {
+            offset = Cids._writeUvarint(buffer, offset, values[i]);
+        }
+
+        // Read all values and verify
+        uint256 currentOffset = 0;
+        for (uint256 i = 0; i < values.length; i++) {
+            (uint256 readValue, uint256 newOffset) = Cids._readUvarint(buffer, currentOffset);
+            assertEq(readValue, values[i], "Uvarint round trip failed");
```

Comment:
Maybe writing everything to one buffer and then reading it out again is a common pattern but I don't think I've seen it before.  It is a nice way to exercise the offset return as well.